### PR TITLE
Store logs from controllers in E2E tests

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Gather logs from scheduler, kcm and kueue controllers for debugging.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: hack/kind-cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  - |
+    kind: ClusterConfiguration
+    apiVersion: kubeadm.k8s.io/v1beta3
+    scheduler:
+      extraArgs:
+        v: "2"
+    controllerManager:
+      extraArgs:
+        v: "2"
+- role: worker
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply hack/kind-cluster.yaml:15]: We do. When we create a node with `role: control-plane`, this node has a taint. Then other pods (like the job) cannot be scheduled.

[the remark this answers is not in the document]
Comment (tenzen-y) [reply hack/kind-cluster.yaml:15]: I see. Thanks!

**Pull request merged**